src/main/java/com/moorhuhnservice/moorhuhnservice/service/AuthorizationService.java

```diff
@@ -0,0 +1,28 @@
+package com.moorhuhnservice.moorhuhnservice.service;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class AuthorizationService {
+
+  public boolean isAuthorized(final String tokenCookie) {
+    boolean authorized = false;
+    try {
+      final Algorithm algorithm = Algorithm.HMAC256("test"); //use more secure key
+      final JWTVerifier verifier = JWT.require(algorithm).build(); //Reusable verifier instance
+      final DecodedJWT jwt = verifier.verify(tokenCookie);
+      authorized = true;
+      log.debug("verification successfully! id was: {}", jwt.getClaim("id"));
+    } catch (JWTVerificationException exception) {
+      log.debug("verification not successfully.", exception);
+    }
+    return authorized;
```

Comment:
```suggestion
    try {
      final Algorithm algorithm = Algorithm.HMAC256("test"); // TODO: use more secure key
      final JWTVerifier verifier = JWT.require(algorithm).build(); //Reusable verifier instance
      final DecodedJWT jwt = verifier.verify(tokenCookie);
      log.debug("verification successfully! id was: {}", jwt.getClaim("id"));
      return true;
    } catch (JWTVerificationException exception) {
      log.info("verification not successfully.", exception);
    }
    return false;
```
The only question is: Should we also show the unverified token in the logs, or would that be a security risk?
```suggestion
    try {
      final Algorithm algorithm = Algorithm.HMAC256("test"); // TODO: use more secure key
      final JWTVerifier verifier = JWT.require(algorithm).build(); //Reusable verifier instance
      final DecodedJWT jwt = verifier.verify(tokenCookie);
      log.debug("verification successfully! id was: {}", jwt.getClaim("id"));
      return true;
    } catch (JWTVerificationException exception) {
      log.info("verification not successful for token {}. Exception: {}", tokenCookie, exception);
    }
    return false;
```

Reply:
This should be an extra issue implementing the security thing right. I mean it is not correct to implement it everywhere in the controller, it has to be anywhere in a SecurityConfiguration in spring. Also, we need roles that for example only a lecturer can edit configurations

Reply:
I recommend taking a look at `interceptors`. They are the concept you want.